packages/block-library/src/image/index.php

```diff
@@ -22,12 +22,33 @@ function render_block_core_image( $attributes, $content, $block ) {
 		return '';
 	}
 
-	$p = new WP_HTML_Tag_Processor( $content );
+	$output = '';
 
-	if ( ! $p->next_tag( 'img' ) || ! $p->get_attribute( 'src' ) ) {
+	/*
+	 * Prior to WP 6.9, WP_HTML_Processor::serialize_token() was `protected`.
+	 * We define an adhoc class to expose that method.
+	 */
+	$internal_processor_class = new class( '', WP_HTML_Processor::CONSTRUCTOR_UNLOCK_CODE ) extends WP_HTML_Processor {
+		// phpcs:ignore Gutenberg.NamingConventions.ValidBlockLibraryFunctionName.FunctionNameInvalid, Gutenberg.Commenting.SinceTag.MissingMethodSinceTag
+		public function serialize_token(): string {
+			return parent::serialize_token();
+		}
+	};
+
+	$p = $internal_processor_class::create_fragment( $content );
```

Comment:
this looks like existing code, but it would be nice to rename `$p` here to `$processor` for consistency with other code in Core as example code.

Reply:
I'll tackle this in a follow-up.

Reply:
https://github.com/WordPress/gutenberg/pull/72033